Reflections/Week_5_reflection.md

```diff
@@ -0,0 +1,28 @@
+## Accomplishments
+
+I have successfully added in basic file I/O to my program.
+
+It is imperfect, I am still not serializing all of the features of my lines as-is and more work will need to be done to support that, however I am happy with being able to save and load at all.
+
+## Challenges
+
+The biggest challenge so far has been in the fact that I could not find a in-built way in the C++ stdlib to write formatted binary output.
+
+It would perhaps be easy to assume that the concept of formatted binary is meaningless, however that is far from the case. Among the considerations for binary output are the following:
+
+- Endianess
+- Byte width of specific elements (sizeof(int) is not specified)
+- null termination for strings or length encoding?
```

Comment:
This is definitely one of those "grey" areas; both approaches are perfectly valid. Null termination requires O(n) to determine length, whereas length encoding is O(1), but may use a few extra bytes per string, depending on the integral type you use to specify the length. It may be a case of "pick one and stick with it".